firmware/hackrf_usb/usb_api_transceiver.c

```diff
@@ -247,6 +250,15 @@ transceiver_mode_t transceiver_mode(void) {
 	return _transceiver_mode;
 }
 
+static void set_m0_mode(enum m0_mode mode) {
+	// All values of the threshold setting are valid. So setting a mode,
+	// it's necessary to set up a transition back to the same mode, for
+	// when the M0 counter wraps.
+	m0_state.mode = mode;
```

Comment:
In order to avoid race conditions, perhaps the M4 should never write to `mode` directly. There could be a separate `requested_mode` or maybe the existing `next_mode` could be used (either by setting `threshold` to a time in the near future or by adding a flag that indicates that the mode should be changed as soon as possible).

Reply:
Fixed in cb3cfe8